api/src/models/task_id.rs

```diff
@@ -0,0 +1,91 @@
+/*-
+ * ========================LICENSE_START=================================
+ * PREvant REST API
+ * %%
+ * Copyright (C) 2018 - 2019 aixigo AG
+ * %%
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * =========================LICENSE_END==================================
+ */
+use http_api_problem::{HttpApiProblem, StatusCode};
+use rocket::http::RawStr;
+use rocket::request::FromParam;
+use std::str::FromStr;
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct TaskId(uuid::Uuid);
+
+impl TaskId {
```

Comment:
Here I'm wondering if we should call it `DeploymentId` because task is soooo generic. And that makes me wondering if the route should be `/deployment/{deploymentId}` instead. What do you think?

Reply:
I've got a slight preference for "task". I associate "deployment" either with something longer-lasting, in the sense that k8s uses the term, or with "the act of deploying something", in which case it would be confusing to poll a deployment when undeploying an app.

But there probably *is* a better term than "task", I just can't find one that's less generic but would still match both deploying and undeploying an app?